src/code_transformation/util.jl

```diff
@@ -91,6 +108,21 @@ function remove_vararg(typ::Expr)
     if isa_vararg(typ)
         body = get_body(typ)
         new_typ = replace_body(typ, body.args[2])
+
+        # This is a bit ugly:
+        # handle interally `where N` from `typ = :(Vararg{FOO, N} where N)` which results in
+        # `body = :(Vararg{FOO, N})` and `new_type = Foo where N`, we don't need to keep it
+        # at all, the `where N` wasn't doing anything to begin with, so we just strip it out
+        if Meta.isexpr(new_typ, :where, 2) && Meta.isexpr(body, :curly, 3)
+            @assert body.args[1] == :Vararg
+            T = body.args[2]
+            N = body.args[3]
+            if new_typ.args == [T, N]
+                body = :(Vararg{T})
+                new_typ = T
+            end
+        end
```

Comment:
Looks like i didn't add any test for this either.
Will need to fix that

Reply:
Ok I have fixed that case,
and added a broken test case for `Vararg{T, N} where T where N`

The only reason `Nabla.remove_vararg(:(Vararg{Real, N} where {N}))` comes up is because ExprTools doesn't do implict unspecified type parameters. No human would do that. As evidenced by not finding this issue before this PR.

And ExprTools it will never make `Vararg{T, N} where T where N` bcause it does believe in collapsing `where` clauses.
(and humans both will never insert unneed and locally scoped `where N` *and* will never chain where clauses to do so).